kernel: bump 5.15 to 5.15.137

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.15.137

All patches automatically rebased.

Build system: x86_64
Build-tested: ramips/tplink_archer-a6-v3
Run-tested: ramips/tplink_archer-a6-v3

DragonBluep commented on target/linux/bcm53xx/patches-5.15/180-usb-xhci-add-support-for-performing-fake-doorbell.patch:
```diff
@@ -90,33 +90,38 @@ it on BCM4708 family.
++	-if (xhci_start(xhci)) {
++	-	xhci_halt(xhci);
++	-	spin_unlock_irqrestore(&xhci->lock, flags);
++	-	return -ENODEV;
```
Perhaps you can copy this patch from `patch-6.1` instead of refreshing it.

graysky2 replied:
I did so but the resulting patch needed a manual rebase.  I am finishing the update now and will force push.

DragonBluep commented on target/linux/bcm53xx/patches-5.15/180-usb-xhci-add-support-for-performing-fake-doorbell.patch:
```diff
@@ -90,41 +90,22 @@ it on BCM4708 family.
-+			goto err_halt;
++		int err = xhci_fake_doorbell(xhci, 1);
++		if (err) {
++			xhci_halt(xhci);
```
It seems that we need `spin_unlock_irqrestore(&xhci->lock, flags);` here? Not sure. Same on kernel 6.1.

graysky2 replied:
I cannot say... someone with more knowledge will need to help out.

PolynomialDivision replied:
Maybe ping Rafał Miłecki?

graysky2 replied:
@rmilecki - can you help out with this rebase?

hauke replied:
You probably wanted to ping @rmilecki 

The patch uses usleep_range(), you can not call that in atomic context like a spinlock.

graysky2 replied:
Rafal mentioned in personal email that he was quite busy but will try to get to this sometime this week.  Meanwhile, I pushed the 5.15.137 update.

graysky2 committed:
kernel: bump 5.15 to 5.15.137

Changelog: https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.15.137

All patches automatically rebased.

Build system: x86_64
Build-tested: ramips/tplink_archer-a6-v3
Run-tested: ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>

rmilecki commented:
Please rebase this on top of the commit c25c1e28b778 ("bcm53xx: backport XHCI patch modifying xhci_run_finished()") and just `rm target/linux/bcm53xx/patches-5.15/081-xhci-Keep-interrupt-disabled-in-initialization-until.patch` while updating 5.15 kernel.

graysky2 commented:
@rmilecki - I see you pulled the commit of the 5.15.136 bump.  Should I remove that from this PR or will someone just cherrypick the 5.15.137 commit and close the PR?  It cannot be merged as-is due to a conflict.

rmilecki commented:
Thanks for handling remaining patch @dangowrt !
